access/validator.go

```diff
@@ -356,14 +356,34 @@ func (v *TransactionValidator) checkSignatureFormat(tx *flow.TransactionBody) er
 
 func (v *TransactionValidator) checkSufficientFlowAmountToPayForTransaction(tx *flow.TransactionBody) error {
 	tracer := tracing.NewMockTracerSpan()
-	systemContracts := environment.NewSystemContracts(
+
+	runtimeParams := environment.DefaultRuntimeParams()
+	runtime := environment.NewRuntime(runtimeParams)
+
+	envParams := environment.DefaultEnvironmentParams()
+	envParams.Chain = v.chain
+
+	// TODO: replace with a real context (have to pass it to validator). it is possible!
+	ctx := context.Background()
+
+	var txState storage.TransactionPreparer
+	// TODO: how to init txState?
```

Comment:
Would it be fine to extend `VM interface` with `CheckPayerBalanceAndGetMaxTxFees`? 
Then I could use it in `QueryExecutor` and `access.ScriptExecutor` which are already set up properly on AN

Reply:
I pushed a commit to express this idea: https://github.com/onflow/flow-go/pull/6004/commits/5efdd5f69f67ddd8e23765b27f210b3234281dec

Reply:
Using the system contracts object seems to be overly complex, so I think it makes more sense to just execute a script here. The validator would have a `ScriptExecutor` object, and use a standard script body.